v2/pkg/protocols/http/signer/aws-sign.go

```diff
@@ -0,0 +1,137 @@
+package signer
+
+import (
+	"bytes"
+	"context"
+	"crypto/sha256"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	v4 "github.com/aws/aws-sdk-go-v2/aws/signer/v4"
+	awsconfig "github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/credentials"
+)
+
+// AWSOptions
+type AWSOptions struct {
+	AwsID          string
+	AwsSecretToken string
+	Service        string
+	Region         string
+}
+
+// Validate Signature Arguments
+func (a *AWSOptions) Validate() error {
+	if a.Service == "" {
+		return errors.New("aws service cannot be empty")
+	}
+	if a.Region == "" {
+		return errors.New("aws region cannot be empty")
+	}
+
+	return nil
+}
+
+// AWS v4 signer
+type AWSSigner struct {
+	creds   *aws.Credentials
+	signer  *v4.Signer
+	options *AWSOptions
+}
+
+// SignHTTP
+func (a *AWSSigner) SignHTTP(ctx context.Context, request *http.Request) error {
+
+	if region, ok := ctx.Value(SignerArg("region")).(string); ok && region != "" {
+		a.options.Region = region
+	}
+	if service, ok := ctx.Value(SignerArg("service")).(string); ok && service != "" {
+		a.options.Service = service
+	}
+	if err := a.options.Validate(); err != nil {
+		return err
+	}
+
+	// Bug Request has content-length but no body
+	rbody, _ := io.ReadAll(request.Body)
+	request.Body.Close()
+	request.Body = io.NopCloser(bytes.NewBuffer(rbody))
+	fmt.Printf("request body is :\n*=\n%v\n=*\n", string(rbody))
+	//
+
```

Comment:
@Mzack9999 

Request passed to `SignHTTP` function is malformed which is core problem for malformed signatures
content length is available but body is empty . However `debug`request shown is correct and is different from what was passed to `SignHTTP` .  

Possible Explaination `might be related to ( generated.Request -> http.Request )`

This can be verified by running Unit Test in `signer` package where request is created using `http.request` and it executes successfully.

Reply:
Above code is only used to Validate the bug